src/pipeline.rs

```diff
@@ -397,41 +371,69 @@ impl Device {
             shader
         };
 
+        Ok(Shader(shader))
+    }
+
+    /// Create a new shader from GLSL.
+    ///
+    /// # Valid usage
+    ///
+    /// - `source` must be a NULL-terminated C-String.
+    /// - The GLSL shader version must be `450 core` or higher.
+    /// - The `stage` parameter must be a valid stage of the passed shader source.
+    pub unsafe fn create_shader(&self, stage: ShaderStage, source: &[u8]) -> Result<Shader> {
```

Comment:
I feel a `ShaderFlag` bitflag parameter would be appropriate here with an option for `VERBOSE` output - so we only have one path for shader creation